src/ProfileEndpoints.jl

```diff
@@ -117,17 +130,29 @@ function _start_cpu_profile(n, delay)
     return resp
 end
 
+function handle_cpu_profile_stop(qp, stage_path = nothing)
+    with_pprof = parse(Bool, get(qp, "pprof", default_pprof()))
+    filename = stage_path === nothing ? "cpu_profile" : joinpath(stage_path, "cpu_profile")
+    # Defer the potentially expensive profile symbolication to a non-interactive thread
+    return fetch(Threads.@spawn _cpu_profile_response($filename; with_pprof=$with_pprof))
+end
+
 function _cpu_profile_response(filename; with_pprof::Bool)
+    # if the parent directory is not set it means that this profile
+    # won't be written to stage and we should inline its content in the
+    # HTTP response
+    parent_dir = dirname(filename)
+    should_inline_profile = isempty(parent_dir)
```

Comment:
🤔 why can't this also be passed through as a separate param? I think that would be clearer, rather than depending on the content of the filename. (For example, what if we were testing this, and wanted to just write the results right to the cwd.)

Reply:
Applied.